app/src/app/api/v0/chat/[inventory]/route.ts

```diff
@@ -0,0 +1,277 @@
+import { HfInference } from "@huggingface/inference";
+import { z } from "zod";
+import { HuggingFaceStream, OpenAIStream, StreamingTextResponse } from "ai";
+import { experimental_buildOpenAssistantPrompt } from "ai/prompts";
+import OpenAI from "openai";
+import { ChatCompletionMessageParam } from "openai/resources/chat/completions";
+import { apiHandler } from "@/util/api";
+import UserService from "@/backend/UserService";
+import { db } from "@/models";
+import { Inventory } from "@/models/Inventory";
+import { Population } from "@/models/Population";
+
+let Hf: HfInference, openai: OpenAI;
+
+function setupModels() {
+  if (!Hf) {
+    Hf = new HfInference(process.env.HUGGINGFACE_API_KEY);
+  }
+  if (!openai) {
+    openai = new OpenAI({
+      apiKey: process.env.OPENAI_API_KEY,
+    });
+  }
+}
+
+// export const runtime = "edge";
+
+const roleSchema = z.enum([
+  "function",
+  "data",
+  "system",
+  "user",
+  "assistant",
+  "tool",
+]);
+const messagesSchema = z.array(
+  z.object({ content: z.string(), role: roleSchema }),
+);
+const chatRequest = z.object({
+  messages: messagesSchema,
+});
+type Messages = z.infer<typeof messagesSchema>;
+
+// Function below is not being referenced/implemented
```

Comment:
Feel free to remove the function instead of commenting it out. It was mainly used for debugging as you can't directly print out the OpenAI response to the console (since it's a stream).

If we need it again, we can find it in the git history.

Reply:
Thanks and done